torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4570,10 +4570,20 @@ def _smart_symbol_sort(x):
                 # references u2 and u3 prior to them actually being bound at
                 # runtime.  It's pretty inconvenient to setup control
                 # dependencies for substitutions, so ban it entirely.
-                if isinstance(lhs, sympy.Symbol) and free_unbacked_symbols(lhs) and not free_unbacked_symbols(rhs):
-                    # short-circuit when no solving is needed
+                def trivial_solve(lhs, rhs):
+                    if isinstance(lhs, sympy.Symbol):
+                        if free_unbacked_symbols(lhs) and not free_unbacked_symbols(rhs):
+                            return True
+                        if symbol_is_type(lhs, SymT.FLOAT):
```

Comment:
do we need any sort of type check on the rhs?

Reply:
No, I think lhs being a float guarantees rhs is float

Reply:
Ah, yes, writing asserts in comments. Classic :P